### ISSUE TITLE
Make it possible to disable smooth scrolling in a buffer

### ISSUE THREAD

**Issue by Hubro** [Kethku/neovide#813]
When I run certain commands in an embedded terminal that produces a *huge* amount of output, the smooth scrolling feature will throw a fit and basically freeze the window for a few seconds.

It would be neat if it was possible to disable the smooth scrolling feature on a per-buffer basis. That way I could use an autocmd to disable smooth scrolling when I enter insert mode in an embedde terminal.

As far as I know, the smooth scrolling feature can't currently be toggled at runtime at all, as it's automatically enabled if Neovide is run with `--multiGrid`.

**Comment by Kethku** [comment]
are you sure that its smooth scrolling? Its possible that the "fit" you're talking about is really a performance issue in shaping/rendering. Can you try something that does this without multigrid enabled? I'm very curious if you run into a similar problem there

**Comment by Kethku** [comment]
that said, I do think this is a reasonable request. Unfortunately disabling the feature per buffer is blocked on https://github.com/Kethku/neovide/issues/720

**Comment by Hubro** [comment]
> are you sure that its smooth scrolling? Its possible that the "fit" you're talking about is really a performance issue in shaping/rendering. Can you try something that does this without multigrid enabled? I'm very curious if you run into a similar problem there

Good point. I re-rested it without multigrid, and the issue went away. It's the output of a `cp -v -r` command on a directory with quite a lot of files, so it instantly dumps a few thousand lines to the terminal. Without multigrid, this finishes immediately and the script immediately continues. With multigrid, the window freezes for a few seconds, then continues.

That said, the smooth scrolling effect on regular, "sane" terminal output is really, really nice. It's probably nice enough that I'll keep multigrid enabled in spite of the freezing issue :smile: 

To reproduce this, you can just run `cp -r -v` on any reasonably large project. The Linux kernel should definitely do it :smile: 

Is it a possible optimization to detect when a massive amount of text is being generated and "skip" the animation?

**Comment by Kethku** [comment]
I'm not sure, but maybe. I'm honestly confused why smooth scrolling would freeze in the way you describe. I'll need to do some investigation. Its possible that the system is taking too many snapshots which is causing the slowdown...

**Comment by Kethku** [comment]
The core issue is fixed in current main, but the request is still reasonable. Many of our settings should be observed at a window level as well as the global setting. Our setting system needs to be updated to track window level information as well
